### PR TITLE
feat: add support for reclient

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This isn't fully ready for public consumption but putting it up now for others to see.  Some missing pieces:
* ~~A more developer friendly way of getting auth onto the machine~~ `e d rbe login`
* ~~Either a more generic hack for https://github.com/bazelbuild/reclient/issues/27 or a real fix upstream~~ Moved TLS termination out of nginx and instead the RBE frontend
* No idea if this even remotely works on windows
* Needs some changes to `e/e` as well https://github.com/electron/electron/pull/40850

### CODE REVIEW

File: src/download.js [hunk 1]
```diff
@@ -8,26 +8,30 @@ const pipeline = promisify(stream.pipeline);
 
 const { fatal } = require('./utils/logging');
 
-const MB_BYTES = 1025 * 1024;
+const MB_BYTES = 1024 * 1024;
```
Comment (VerteDinde) [inline src/download.js:11]: Just curious, why the bump from 1025 -> 1024 here?

**Review by VerteDinde — APPROVED** [review]

File: src/utils/reclient.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+const reclientDir = path.resolve(__dirname, '..', '..', 'third_party', 'reclient');
+const reclientTagFile = path.resolve(reclientDir, '.tag');
+const reclientHelperPath = path.resolve(
+  reclientDir,
+  `electron-rbe-credential-helper${process.platform === 'win32' ? '.exe' : ''}`,
+);
+const rbeServiceAddress = 'rbe.notgoma.com:443';
+
+const CREDENTIAL_HELPER_TAG = 'v0.1.0';
```
Comment (dsanders11) [inline src/utils/reclient.js:17]:
```suggestion
const CREDENTIAL_HELPER_TAG = 'v0.2.0';
```

Newest?

File: src/download.js (hunk 1 above)
Comment (MarshallOfSound) [reply src/download.js:11]: A megabyte is 1024 * 1024, idk why it was 1025 before, that was just wrong afaics

File: src/utils/reclient.js (hunk 2 above)
Comment (MarshallOfSound) [reply src/utils/reclient.js:17]: The `0.2.0` credential helper didn't publish correctly because of a codesigning issue

**Pull request merged**